### PR TITLE
feat: add editable button

### LINKED ISSUE #4: Update Todo
What I want is to update a todo in the list by clicking on it so it becomes an input and when I click out of it (or in another way lose focus 😉 ) I should see that it returned to a normal todo text. 

Now you need some new tools in your belt to achieve this task. First, you need to know about a new event in the [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) called as onBlur (which is the opposite of onFocus) to trigger the change.

At this event, you should map todos so that only the clicked one is converted to an `input`. And do the same when you blur and convert it to normal text. I see that this is not a beginner-friendly task. However, I believe that it will give you a myriad of new skills that you will need along the way.

Image

### PR DESCRIPTION
**Pull request by mmusta**
closes #4

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -2,18 +2,26 @@
 import { useRef, useState } from 'react';
 
 function App() {
+  const [isEditableList, setIsEditableList] = useState([]);
```
Comment (berakoc) [inline src/App.jsx:5]: naming is always important.

File: src/App.jsx [hunk 2]
```diff
@@ -2,18 +2,26 @@
...
   const inputRef = useRef(null);
   const handleAddTodoClick = () => {
     const newTodo = inputRef.current.value;
     setTodos(todos.concat(newTodo));
     inputRef.current.value = '';
+    setIsEditableList(isEditableList.concat(false));
   };
   const handleDeleteClick = (index) => {
-    const newTodos = todos.filter((_, todoIndex) => todoIndex !== index);
+    const newTodos = [...todos.slice(0, index), ...todos.slice(index + 1)];
```
Comment (berakoc) [inline src/App.jsx:15]: you really achieved great progress there. some parts were missing but this is just a beginning

File: src/App.jsx [hunk 3]
```diff
@@ -26,8 +34,9 @@ function App() {
       <ul>
         {todos.map((todo, index) => (
           <li key={todo}>
-            {todo}
+            {isEditableList[index] ? <input /> : <span>{todo}</span> }
```
Comment (berakoc) [inline src/App.jsx:37]: Now you learned a myriad of concepts, namely conditional rendering, using refs for manipulating DOM nodes and manipulation state in an immutable way which is always the standard.

**Review by berakoc — APPROVED** [review]

**Comment by mmusta** [comment]
closes #4

**Pull request merged**